File: oop/practiseC/revision/freq_str.c
#include<stdio.h>

void main()
{
	char str[50],key;
	int i,count=0, len;
	printf("Enter the string : \n");
	scanf(" %s",str);
	printf("Enter key to search \n");
	scanf(" %c",&key);
	for(len = 0;str[len]!=0;len++);
	for ( i = 0; i < len; ++i)
	{
		if (str[i]==key)
		{
			count++;
		}
	}
	printf("%c occured %d times\n",key,count);
}

File: oop/cycle1/FreqStr.java
/** Java program to check the frequency of a given character in a string which is entered by the user */

import java.util.Scanner;

public class FreqStr{
	public static void main(String[] args){
		String str;
		char key;
		int count = 0, i,len;
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter string ");
		str  = sc.nextLine();
		System.out.println("Enter key to search ");
		key = sc.next().charAt(0);
		len = str.length();
		for (i = 0;i<len;i++) {
			if (key == str.charAt(i)) {
				count ++;
			}
		}
		System.out.println("count is "+count);
	}
}